hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/store/FederationDelegationTokenStateStore.java

```diff
@@ -114,37 +114,16 @@ RouterRMTokenResponse getTokenByRouterStoreToken(RouterRMTokenRequest request)
       throws YarnException, IOException;
 
   /**
-   * The Router Supports incrementDelegationTokenSeqNum.
+   * Return a new unique integer lookup key for a new delegation token
    *
    * @return DelegationTokenSeqNum.
    */
-  int incrementDelegationTokenSeqNum();
+  int getNewDelegationTokenKey();
 
   /**
-   * The Router Supports getDelegationTokenSeqNum.
-   *
-   * @return DelegationTokenSeqNum.
-   */
-  int getDelegationTokenSeqNum();
-
-  /**
-   * The Router Supports setDelegationTokenSeqNum.
-   *
-   * @param seqNum DelegationTokenSeqNum.
-   */
-  void setDelegationTokenSeqNum(int seqNum);
-
-  /**
-   * The Router Supports getCurrentKeyId.
-   *
-   * @return CurrentKeyId.
-   */
-  int getCurrentKeyId();
```

Comment:
have removed these methods because they aren't required for a stateless secret manager. they are only required to support recovery in RM / NN like systems with only 1 node generating tokens . 

with stateless secret manager, there is no explicit recovery mechanism because by design the secret manager stores all data in database and thus these methods aren't required.